core/src/main/java/org/elasticsearch/index/Index.java

```diff
@@ -31,21 +32,23 @@
 public class Index implements Streamable {
 
     private String name;
+    private String uuid;
 
     private Index() {
 
     }
 
-    public Index(String name) {
+    public Index(String name, String uuid) {
```

Comment:
can we please get rid of the interning? I think we should do this in a sep PR before this get's merged